vizro-core/src/vizro/models/_controls/filter.py

```diff
@@ -96,19 +96,25 @@ def check_target_present(cls, target):
 
     @_log_call
     def pre_build(self):
-        if self.targets:
-            targeted_data = self._validate_targeted_data(targets=self.targets)
-        else:
-            # If targets aren't explicitly provided then try to target all figures on the page. In this case we don't
-            # want to raise an error if the column is not found in a figure's data_frame, it will just be ignored.
-            # Possibly in future this will change (which would be breaking change).
-            targeted_data = self._validate_targeted_data(
-                targets=model_manager._get_page_model_ids_with_figure(
-                    page_id=model_manager._get_model_page_id(model_id=ModelID(str(self.id)))
-                ),
-                eagerly_raise_column_not_found_error=False,
-            )
-            self.targets = list(targeted_data.columns)
+        # If targets aren't explicitly provided then try to target all figures on the page. In this case we don't
+        # want to raise an error if the column is not found in a figure's data_frame, it will just be ignored.
+        # This is the case when bool(self.targets) is False.
+        # Possibly in future this will change (which would be breaking change).
+        proposed_targets = self.targets or model_manager._get_page_model_ids_with_figure(
+            page_id=model_manager._get_model_page_id(model_id=ModelID(str(self.id)))
+        )
+        # TODO NEXT: how to handle pre_build for dynamic filters? Do we still require default argument values in
+        #  `load` to establish selector type etc.? Can we take selector values from model_manager to supply these?
+        #  Or just don't do validation at pre_build time and wait until state is availableduring build time instead?
+        #  What should the load kwargs be here?
+        #  Note that currently _get_unfiltered_data is only suitable for use at runtime since it requires
+        #  ctd_parameters. That could be changed to just reuse that function.
```

Comment:
I'll test this case after this initial review and will add conclusions in this thread.